### PR TITLE
Add tests for handling invalid UTF-8 insertion strings in WelSupportedTest

### PR DESCRIPTION
**Pull request by Boombag0607**
## Description

This pull request addresses issue #273. Adds new tests to the `WelSupportedTest` fixture in order to handle invalid UTF-8 insertion strings. The tests focus on various scenarios to ensure correct behaviour and error handling.

### Test Cases

1. `InvalidUtf8InsertionString`: This test verifies that an error is raised when an invalid UTF-8 string is provided as an insertion string.

2. `InsertionIndexNotModifiedWhenInvalidUtf8StringProvided`: This test checks that the insertion string at the specified index remains unchanged when an invalid UTF-8 string is provided.

3. `InvalidUtf8InsertionStringOutOfRange`: This test ensures that an error is raised when attempting to set an invalid UTF-8 string at an out-of-range index.

4. `InsertionIndexNotModifiedWhenOutOfRangeIndexProvided`: This test verifies that the insertion string at a valid index remains unchanged when an out-of-range index is provided for an invalid UTF-8 string.

5. `InsertionIndexNotModifiedWhenInRangeButNotAssigned`: This test confirms that an unassigned insertion string at a valid index returns `nullptr`.

### CODE REVIEW

**Comment by goatshriek** [comment]
I'm converting this to a draft for now just to make it clear that this is still in work. Once you're all finished with it or you need help, let me know and I'll review it.

**Comment by Boombag0607** [comment]
Sure thing. The problem I'm facing is that I'm currently using a Mac M1 and since the test is particularly Windows related, I can't really find a way to compile the files/ run the workflows locally. There's really no way for me to know the checks before pushing. Can you suggest a workaround?

**Comment by goatshriek** [comment]
Without having access to a Windows machine to work on, this will be a little more difficult for you. One possible workaround is to enable Github Actions on your fork, remove all workflows but the Windows ones, and push to it to run the Windows CI workflows when you need to test something. You'd need to be sure to add the original workflow files before the request is marked as ready.

You could also try something like [Wine](https://www.winehq.org/), but the library hasn't been tested on this and I wouldn't be able to offer any support for getting it working. Using Actions on your own fork is probably the easiest thing to do, so that you won't need to wait for me to approve runs on the upstream repository.

File: .github/workflows/cygwin.yml
Comment (goatshriek) [inline .github/workflows/cygwin.yml]: This file shouldn't be changed - restore it to the original version now that your testing is complete.

File: .github/workflows/windows.yml
Comment (goatshriek) [inline .github/workflows/windows.yml]: This file should also be restored to the original so that it isn't included in this change.

File: test/function/config/wel_supported.cpp
Comment (goatshriek) [inline test/function/config/wel_supported.cpp]: This line and a few others are over 80 characters long: reformat where you can to bring these to 80 characters or under. There are examples elsewhere in this file of how to format long function calls across multiple lines.

File: test/function/config/wel_supported.cpp
Comment (goatshriek) [inline test/function/config/wel_supported.cpp:347]: Add checks after calls with invalid UTF8 to make sure that the error id is set correctly: it should be `STUMPLESS_INVALID_ENCODING`. See some of the above tests for examples of how to do this using the `EXPECT_ERROR_ID_EQ` macro. Also note that this helper requires a `stumpless_error` struct named "error" to be declared in order to work.

File: test/function/config/wel_supported.cpp
Comment (goatshriek) [inline test/function/config/wel_supported.cpp]: add result and error id checks after this call

**Review by goatshriek — CHANGES_REQUESTED** [review]
Thanks very much for putting these tests together, they are mostly ready! There are a few updates for you to make listed below, after which I will merge this in.

File: .github/workflows/cygwin.yml
Comment (Boombag0607) [reply .github/workflows/cygwin.yml]: Oh that was not intended to happen, must have been merged from a different branch.

File: test/function/config/wel_supported.cpp
Comment (Boombag0607) [reply test/function/config/wel_supported.cpp]: Sure, I'm fixing the same. Can you elaborate on the insertion string being out of range part (from the issue #273) so that I can add this test to the commit as well, I could not get how a valid insertion string be out of range and the invalid one being in range.

Comment (goatshriek) [reply test/function/config/wel_supported.cpp]: The language in the issue is a little confusing, sorry about that. You can achieve the state I was trying to describe by setting a valid insertion string at a high index (I would use index 3 or 4, since the fixture entry already has 2 insertion strings), and then setting the invalid one at index 2. Index 2 won't have been assigned yet, but will be in the already allocated range because of the higher string successfully set. Your test will need to make sure that it is still empty after the attempt to set it to an invalid string.

Comment (Boombag0607) [reply test/function/config/wel_supported.cpp]: Got it.

**Review by goatshriek — APPROVED** [review]

**Pull request merged**
